### PR TITLE
Optional limit

### PR DESCRIPTION
**Pull request by alex-taxiera**
`limit` is an optional parameter to `select` currently, if no limit is passed, we count all entries in the table and set limit to the count. That is pretty unnecessary when we could just not use limit if we don't need to.

### CODE REVIEW

File: src/QueryBuilder.js [hunk 1]
```diff
@@ -312,35 +312,36 @@ class QueryBuilder {
...
-              const old = rows[i][key]
-              rows[i][key] = JSON.parse(old)
-              if (typeof rows[i][key] === 'number') {
-                rows[i][key] = old
-              }
-            } catch (e) {
-              continue
+
+    if (limit) {
+      query.limit(limit)
```
Comment (mets11rap) [inline src/QueryBuilder.js]: Instead of storing the query to a variable, what if we do `.limit(limit || Infinity)`?

Comment (GitStonic) [reply src/QueryBuilder.js]: should have a fixed limit...

Comment (mets11rap) [reply src/QueryBuilder.js]: Why? That would be limiting to big services.

Comment (GitStonic) [reply src/QueryBuilder.js]: nevermind

Comment (alex-taxiera) [reply src/QueryBuilder.js]: looks like knex works with that

**Pull request merged**